CheckManagement/LocalRepository.cs

```diff
@@ -26,11 +29,85 @@ public class LocalRepository : IRepository
         /// </summary>
         private string FolderPath { get; }
 
+        /// <summary>
+        /// A default repository name to use when one is not provided.
+        /// </summary>
+        private string _defaultName = MainConsts.LOCAL_REPO_NAME;
+
         public LocalRepository(string folderPath)
         {
             FolderPath = folderPath;
         }
 
+        /// <summary>
+        /// A unique identifier for the repository.
+        /// </summary>
+        public Guid Id { get { return new Guid(); } }
+
+        /// <summary>
+        /// The display name for the repository.
+        /// </summary>
+        public string Name
+        {
+            get 
+            {
+                // The local repository cannot be renamed.
+                return _defaultName;
+            }
+
+            set { }
+        }
+
+        /// <summary>
+        /// Determines if this repository is enabled
+        /// </summary>
+        public bool Enabled
+        {
+            get
+            {
+                // The local repository cannot be disabled.
+                return true; 
+            }
+
+            set { }
```

Comment:
They are necessary because LocalRepository implements IRepository which requires a set clause. IRepository is also implemented by S3Repository which needs the ability to change the Enabled property.

Reply:
Per your suggestion I have changed the setter on these properties to throw an exception instead of failing silently.